Modules/LearningSequence/classes/LearnerProgress/class.ilLSLPEventHandler.php

```diff
@@ -46,12 +55,19 @@ public function updateLPForChildEvent(array $parameter)
      * @return int | false;
      */
     protected function getParentLSO(int $child_ref_id)
+    {
+        if (!array_key_exists($child_ref_id, $this->cached_parent_lso)) {
+            $this->cached_parent_lso[$child_ref_id] = $this->getParentLSOByFullPath($child_ref_id);
+        }
+        return $this->cached_parent_lso[$child_ref_id];
+    }
+
+    private function getParentLSOByFullPath(int $child_ref_id)
```

Comment:
@nhaagen One more thought: AFAIK the LSO can only contain leaf nodes (meaning: non-containers)

So if the learning progress state of a user for an arbitrary object is updated, only the direct parent should be relevant when checking for a LSO. So maybe `\ilTree::getParentNodeData` is more appropriate as `\ilTree::getPathFull`.